src/Proto.Mailbox/Mailbox.cs

```diff
@@ -115,10 +115,9 @@ private Task RunAsync()
                     _stats[i].MailboxEmpty();
                 }
             }
-            return Task.FromResult(0);
         }
 
-        private bool ProcessMessages()
+        private async Task<bool> ProcessMessages()
```

Comment:
Should work :) Tiny improvement:
```
var t = _invoker.EscalateFailureAsync(t.Exception, msg);
if (!t.IsCompleted)
{
    t.ContinueWith(RescheduleOnTaskComplete, msg)
    return false;
}
```

We should probably also check if the task is faulted - but I'm not sure what we can do if that happens. Log a warning?